### ISSUE TITLE
`match()` fails on words prefixed with `{space}{hyphen}{nospace}`

### ISSUE THREAD

**Issue by thegoatherder** [spencermountain/compromise#970]
Words prefixed with a hyphen (`-`) fail to match:

```javascript
const nlp = require('compromise')
console.log(nlp('Please - Match me').match('Match me').out())
console.log(nlp('Please -Match me').match('Match me').out())
```

**Actual Output**
```ascii
Match me
```

**Expected Output**
```ascii
Match me
Match me
```

**Workaround 1**
Pre-process the text with regex to insert space after `-`. Not ideal as the text does not respect the original

**Workaround 2**
Add the hyphen to the match pattern - Not ideal as it (a) adds complexity to pattern writing and; (b) outputs the hypen in the match

```javascript
const nlp = require('compromise')
console.log(nlp(`Please -Match me`).match('-Match me').out())
```

```ascii
-Match me
```

**Comment by spencermountain** [comment]
fixed in `14.6.0` 
cheers

**Comment by thegoatherder** [comment]
Thanks, @spencermountain! 🙏
